cmd/vela-git/repo.go

```diff
@@ -18,6 +18,8 @@ type Repo struct {
 	Submodules bool
 	// enable fetching of tags
 	Tags bool
+	// target branch for pull request
+	PrTargetBranch string
```

Comment:
We'd have to add the field to the `Repo` struct:

```go
	// enables fetching target branch for pull requests
	FetchTarget bool
```

After all of the changes above were complete, you could then opt-in to using this functionality like:

```yaml
version: "1"

metadata:
  # disables the default "clone" process injected by Vela
  clone: false

steps:

  - name: clone
     image: target/vela-git
     parameters:
       fetch_target: true

<more steps in pipeline>
```

Reply:
Sorry for the delay! I appreciate all of this, but the thing I'm worried about is if this is worth it to add as a possible flag for the plugin. 

It's not hard to add support for this manually and it's possibly more code to use this rather than to just add the patch step. I'm not sure how many other static analysis tools perform diffs in this way so the use case is small enough that it may not be worthwhile to merge this functionality. I'm down to implement the changes, close this for later and re-open this later if your team sees that this is worthwhile :+1:

Reply:
> ... the thing I'm worried about is if this is worth it to add as a possible flag for the plugin.

Ultimately, I think that is up to you and your team since I think you folks are the first to report it 👍 

I think from an admin perspective, we don't really have any concerns supporting this in the plugin.

That assumes this functionality is optional due to the extra latency it adds.

> It's not hard to add support for this manually and it's possibly more code to use this rather than to just add the patch step.

I agree that it's not hard to add support for this manually 👍 

> I'm not sure how many other static analysis tools perform diffs in this way so the use case is small enough that it may not be worthwhile to merge this functionality.

I also am unaware of how many other tools would require this kind of functionality 👍 

You could be right that the use case for this could be small enough that it doesn't need to be added to the plugin.